src/parser.coffee

```diff
@@ -302,18 +386,26 @@ exports.Parser = class Parser
           lastIndex = indentDepth
 
         for mark in markupOnLines[i]
+          # If we are not supposed to be flagging to remove,
```

Comment:
I don't understand the flagToRemove mechanism.  The comment here (or elsewhere) needs to explain the situation a bit more clearly.